### PR TITLE
[Concept Entry] Sklearn: Probability Calibration

### LINKED ISSUE #5022: [Concept Entry] Sklearn probability-calibration
### Reason/inspiration (optional)

We would like a new entry on the `Probability Calibration` concept under Sklearn. The entry should go in a new file under `docs/content/sklearn/concepts/probability-calibration/probability-calibration.md`.

The entry should include:

- An introduction to the concept
- A `Syntax` section that provides the syntax for the concept (if applicable)
- An `Example` section that provides an example demonstrating the concept in use (if applicable)
- A `Codebyte Example` section that provides a codebyte example demonstrating the concept in use (if applicable)

Please refer to the [concept entry template](https://github.com/Codecademy/docs/blob/main/documentation/concept-entry-template.md), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Concept Entrty for Existing Topic

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by Sriparno08**
### Description

Added a new term entry on `Probability Calibration` in Sklearn.

### Issue Solved

Closes #5022

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

File: content/sklearn/concepts/probability-calibration/probability-calibration.md [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+from sklearn.linear_model import LogisticRegression
+
+# Create a logistic regression classifier
+model = LogisticRegression()
+
+# Calibrate the classifier using Platt Scaling
+model_calibrated = CalibratedClassifierCV(model, cv=5, method="sigmoid")
+
+# Fit the calibrated classifier to the training data
+model_calibrated.fit(X_train, y_train)
```
Comment (PragatiVerma18) [inline content/sklearn/concepts/probability-calibration/probability-calibration.md:41]: Add a comment to explain what `X_train` and `y_train` are.


```suggestion
# X_train: Features for the training set; y_train: Target labels for the training set
model_calibrated.fit(X_train, y_train)
```

File: content/sklearn/concepts/probability-calibration/probability-calibration.md [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+from sklearn.linear_model import LogisticRegression
+
+# Create a logistic regression classifier
+model = LogisticRegression()
+
+# Calibrate the classifier using Isotonic Regression
+model_calibrated = CalibratedClassifierCV(model, cv=5, method="isotonic")
+
+# Fit the calibrated classifier to the training data
+model_calibrated.fit(X_train, y_train)
```
Comment (PragatiVerma18) [inline content/sklearn/concepts/probability-calibration/probability-calibration.md:60]: Add a comment to explain what `X_train` and `y_train` are.


```suggestion
# X_train: Features for the training set; y_train: Target labels for the training set
model_calibrated.fit(X_train, y_train)
```

File: content/sklearn/concepts/probability-calibration/probability-calibration.md [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+Sklearn provides two primary methods for implementing probability calibration:
+
+- **Platt Scaling**: Fits a logistic regression model to the model's output probabilities.
+- **Isotonic Regression**: Fits a non-parametric isotonic regression model to the model's output probabilities.
+
+## Syntax
+
+The `CalibratedClassifierCV` class is used to implement probability calibration.
+
+Following is the syntax for implementing probability calibration using Platt Scaling:
```
Comment (PragatiVerma18) [inline content/sklearn/concepts/probability-calibration/probability-calibration.md:28]: We should have more information about these methods and how they relate to the method parameter.

```
Platt Scaling uses a `sigmoid` function to map raw model scores to calibrated probabilities, ensuring they better reflect true likelihoods.

The sigmoid function,  
σ(x) = 1 / (1 + e^(-x)),  
maps any real-valued number to a range between 0 and 1.  

In Platt Scaling, this function is parameterized as:  
P(y=1 | x) = 1 / (1 + e^(-(A * x + B))),  

where A and B are parameters learned during calibration.
```

File: content/sklearn/concepts/probability-calibration/probability-calibration.md [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+model_calibrated = CalibratedClassifierCV(model, cv=5, method="sigmoid")
+
+# Fit the calibrated classifier to the training data
+model_calibrated.fit(X_train, y_train)
+
+# Make predictions using the calibrated classifier
+y_pred_prob = model_calibrated.predict_proba(X_test)
+```
+
+Following is the syntax for implementing probability calibration using Isotonic Regression:
```
Comment (PragatiVerma18) [inline content/sklearn/concepts/probability-calibration/probability-calibration.md:47]: We should have more information about how the method parameter relates here and what is Isotonic regression.

```
Isotonic regression is a non-parametric regression technique that fits a piecewise constant, monotonic (increasing or decreasing) function to the data. 

In the context of calibration, the isotonic method uses isotonic regression to map the model's raw probabilities to calibrated probabilities while preserving their relative order.
```

File: content/sklearn/concepts/probability-calibration/probability-calibration.md [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+
+# Fit the calibrated classifier to the training data
+model_calibrated.fit(X_train, y_train)
+
+# Make predictions using the calibrated classifier
+y_pred_prob = model_calibrated.predict_proba(X_test)
+```
+
+- `cv`: The number of cross-validation folds. The default value is `None`.
+- `method`: The calibration method. Common options include `"sigmoid"` (default) and `"isotonic"`.
```
Comment (PragatiVerma18) [inline content/sklearn/concepts/probability-calibration/probability-calibration.md:67]: We should only wrap the exact code terms in backticks.

```suggestion
- `method`: The calibration method. Common options include `sigmoid` (default) and `isotonic`.
```

**Review by PragatiVerma18 — CHANGES_REQUESTED** [review]
Hey @Sriparno08, thanks for contributing this entry, I have few suggestions. Please check the comments and do the required changes. Let me know if you have any questions.

**Comment by Sriparno08** [comment]
Made the changes, @PragatiVerma18.

**Review by PragatiVerma18 — APPROVED** [review]
Hey @Sriparno08, thanks for doing the changes so quickly. This is good to be merged! 🚀
